ruff.toml

```diff
@@ -1,11 +1,50 @@
 line-length = 100
 target-version = "py39"
 
+[lint.pylint]
+max-args = 10
+
 [lint]
+pydocstyle.convention = "google"
 select = [
-    "F", # Pyflakes
-    "E", # pycodestyle
-    "W", # pycodestyle
-    "I", # isort
+    "ALL",
+]
+ignore = [
+    "ANN", # flake8-annotations
+    "C90", # mccabe complexity
+    "PTH", # Will be fixed in INGK-1020
+    "N801", # Will be fixed in INGK-1022
+    "D100", # Missing docstring in public module
+    "D104", # Missing docstring in public package
+    "D107", # Missing docstring in __init__
+    "FA100", # Missing `from __future__ import annotations`
+    "FIX002", # TODO comment
+    "G004", # Logging statement uses f-string
+    "S314", # Insecure XML parser,
+    "SLF001", # Private member accessed
+    "TD002", # Missing author in TODO
+    "TD003", # Missing issue link in TODO
+    "TRY003", # Avoid specifying long messages outside the exception class
+    "COM812", # Trailing comma. Already enforced by the formatter.
+    "ISC001", # Implicitly concatenated strings. Already enforced by the formatter.
+    "S602", # Subprocess call with shell enabled.
+    "S318", # Insecure XML parser.
+    "FBT001", # Boolean positional argument
+    "FBT002", # Boolean positional argument
+]
+[lint.per-file-ignores]
+"ingenialink/network.py" = [
+    "D102", # Missing docstring in public method
+]
+"tests/*" = [
+    "D", # Docstring rules
+    "ERA001", # Commented-out code
+    "S101", # Use of assert keyword
+    "PLR2004", # Magic value comparison
+    "PT004", # Missing fixture name underscore
+    "INP001", # Missing __init__.py file
 ]
-pydocstyle.convention = "google"
\ No newline at end of file
+"virtual_drive/*" = [
+    "D", # Docstring rules
+    "ERA001", # Commented-out code
+]
```

Comment:
Virtual drive should not ignore same rules. Create issues for this if it's too much work